doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -464,39 +461,48 @@ Requests to the environment
       Use the :cmd:`Add @table` and :cmd:`Remove @table` commands to update the set of
       blacklisted strings.
 
-.. cmd:: Locate @qualid
+.. cmd:: Locate @locatable
+
+   .. insertprodn locatable locatable
 
-   This command displays the full name of objects whose name is a prefix
-   of the qualified identifier :n:`@qualid`, and consequently the |Coq| module in
+   .. prodn::
+      locatable ::= @smart_qualid
+      | Term @smart_qualid
+      | Module @qualid
+      | Ltac @qualid
+      | Library @qualid
+      | File @string
+
+   Displays the full name of objects whose name is a prefix
+   of the specified value, and consequently the |Coq| module in
    which they are defined. It searches for objects from the different
    qualified namespaces of |Coq|: terms, modules, Ltac, etc.
 
+   The alternatives have the following behavior:
+
+   -  :n:`Locate @smart_qualid` - displays the full name of objects whose name is a prefix
+      of :n:`@smart_qualid`, thereby showing the module they are defined in.
+   -  :n:`Locate Term @smart_qualid` - limits the search to terms
+   -  :n:`Locate Module @qualid` - limits the search to modules
+   -  :n:`Locate Ltac @qualid` - reports whether the specified module has been loaded
+   -  :n:`Locate Library @qualid` - gives the status of the specified |Coq| module. It reports whether
+      the module is loaded and, if not, searches for the module in the load path and reports
+      where it is found.
+   -  :n:`Locate File @string` - displays the location of a file specified by :n:`@string`
+      Typically, :n:`@string` is a ``.cmo`` or ``.vo`` or ``.v`` file, such as :n:`Nat.v`.
```

Comment:
Same comment than for `Search`. I appreciate the top-level description of what is `locatable`, but then, I would like to see descriptions organized using `cmdv`.

Note that I think we should do the same with `Print`: document all the `printable` in a single command (i.e. consolidating all the `Print` commands that are currently spread out in the manual, putting a link to the right variant where they currently appear) and present each with a separate `cmdv`.

Reply:
I thought an alternative for Locate would to make them all cmds.  It's not clear how much relation there is between them, nor is `locatable` really interesting.  `Locate File`, for example, is clearly different from the others, so why even call it a variant?

> I think we should do the same with `Print`: document all the `printable` in a single command (i.e. consolidating all the `Print` commands that are currently spread out in the manual, putting a link to the right variant where they currently appear) and present each with a separate `cmdv`.

One could create a `Print commands` section that lists and links to all the alternatives that may be elsewhere in the document, perhaps with a couple words describing each.  Consider `Print Foo`: should it be described in one place with all other `Print` variations, or should it be described in the section on `Foo`?

Unless we create some rules about what's in a variant, doc_grammar likely won't be able to verify update them automatically or flag them as needing attention.  That seems like giving up a lot.

Reply:
OK, I am fine with your alternative proposal of gathering all the `Print` / all the `Locate` into a section, but still making them separate `cmd`. I think I prefer to gather them rather than spreading them out (next to the explanation on the associated concept) because it's good to have a reference of all the available queries somewhere and this does not prevent from linking to them from the explanation on the associated concept.

Reply:
All 45 Prints in one section, pulling them out of various chapters?

FYI:
doc.sphinx.addendum
extraction.rst
.. cmd:: Print Extraction Inline
.. cmd:: Print Extraction Blacklist
generalized-rewriting.rst
.. cmd:: Print Instances
implicit-coercions.rst
.. cmd:: Print Classes
.. cmd:: Print Coercions
.. cmd:: Print Graph
.. cmd:: Print Coercion Paths @class @class
doc.sphinx.language
gallina-extensions.rst
.. cmd:: Print Module @qualid
.. cmd:: Print Module Type @qualid
.. cmd:: Print Implicit @smart_qualid
.. cmd:: Print Canonical Projections {* @smart_qualid }
.. cmd:: Print {? Sorted } Universes {? Subgraph ( {* @qualid } ) } {? @string }
doc.sphinx.proof-engine
ltac.rst
.. cmd:: Print Ltac @qualid
.. cmd:: Print Ltac Signatures
tactics.rst
.. cmd:: Print Hint @ident
.. cmd:: Print Hint
.. cmd:: Print Hint @ident
.. cmd:: Print Hint *
.. cmd:: Print HintDb @ident
.. cmd:: Print Rewrite HintDb @ident
   .. cmd:: Print Firstorder Solver
vernacular-commands.rst
.. cmd:: Print {? Term } @smart_qualid {? @univ_name_list }
.. cmd:: Print All
.. cmd:: Print Section @qualid
.. cmd:: Print Options
.. cmd:: Print Table @setting_name
.. cmd:: Print Tables
.. cmd:: Print Assumptions @smart_qualid
.. cmd:: Print Opaque Dependencies @smart_qualid
.. cmd:: Print Transparent Dependencies @smart_qualid
.. cmd:: Print All Dependencies @smart_qualid
.. cmd:: Print Libraries
.. cmd:: Print ML Modules
.. cmd:: Print LoadPath {? @dirpath }
.. cmd:: Print ML Path
.. cmd:: Print Strategy @smart_qualid
.. cmd:: Print Strategies
.. cmd:: Print Typing Flags
doc.sphinx.user-extensions
syntax-extensions.rst
.. cmd:: Print Grammar constr.
.. cmd:: Print Grammar pattern.
.. cmd:: Print Custom Grammar @ident.
.. cmd:: Print Visibility
   .. cmdv:: Print Visibility @scope
.. cmd:: Print Scopes
   .. cmdv:: Print Scope @scope
   .. cmd:: Print Grammar tactic

Reply:
> I think I prefer to gather them rather than spreading them out (next to the explanation on the associated concept) because it's good to have a reference of all the available queries somewhere and this does not prevent from linking to them from the explanation on the associated concept.

This rationale seems a bit weak: all the Prints are already gathered together in the index and hyperlinked to where they're referenced, though there is no terse summary of what each one does.  Does that affect your preference?

Reply:
OK, you are right, that's too much. It's just that it would have made sense to separate the description of what is really the Coq language from all the commands that are just there to request information about the current state. But that can wait.